common/src/main/kotlin/org/neo4j/connectors/kafka/data/PropertyType.kt

```diff
@@ -0,0 +1,369 @@
+          .build()
+
+  fun toConnectValue(value: Any?): Struct? {
+    return when (value) {
+      null -> null
+      is Boolean -> Struct(schema).put(BOOLEAN, value)
+      is Float -> Struct(schema).put(FLOAT, value.toDouble())
+      is Double -> Struct(schema).put(FLOAT, value)
+      is Number -> Struct(schema).put(LONG, value.toLong())
+      is String -> Struct(schema).put(STRING, value)
+      is Char -> Struct(schema).put(STRING, value.toString())
+      is CharArray -> Struct(schema).put(STRING, String(value))
+      is CharSequence ->
+          Struct(schema).put(STRING, value.codePoints().toArray().let { String(it, 0, it.size) })
+      is ByteArray -> Struct(schema).put(BYTES, value)
+      is ByteBuffer -> Struct(schema).put(BYTES, value.array())
+      is LocalDate -> Struct(schema).put(LOCAL_DATE, DateTimeFormatter.ISO_DATE.format(value))
+      is LocalDateTime ->
+          Struct(schema).put(LOCAL_DATE_TIME, DateTimeFormatter.ISO_DATE_TIME.format(value))
+      is LocalTime -> Struct(schema).put(LOCAL_TIME, DateTimeFormatter.ISO_TIME.format(value))
+      is OffsetDateTime ->
+          Struct(schema).put(ZONED_DATE_TIME, DateTimeFormatter.ISO_DATE_TIME.format(value))
+      is ZonedDateTime ->
+          Struct(schema).put(ZONED_DATE_TIME, DateTimeFormatter.ISO_DATE_TIME.format(value))
+      is OffsetTime -> Struct(schema).put(OFFSET_TIME, DateTimeFormatter.ISO_TIME.format(value))
+      is IsoDuration ->
+          Struct(schema)
+              .put(
+                  DURATION,
+                  Struct(durationSchema)
+                      .put(MONTHS, value.months())
+                      .put(DAYS, value.days())
+                      .put(SECONDS, value.seconds())
+                      .put(NANOS, value.nanoseconds()))
+      is Point ->
+          Struct(schema)
+              .put(
+                  POINT,
+                  Struct(pointSchema)
+                      .put(SR_ID, value.srid())
+                      .put(X, value.x())
+                      .put(Y, value.y())
+                      .also {
+                        it.put(DIMENSION, if (value.z().isNaN()) TWO_D else THREE_D)
+                        if (!value.z().isNaN()) {
+                          it.put(Z, value.z())
+                        }
+                      })
+      is ShortArray -> Struct(schema).put(LONG_LIST, value.map { s -> s.toLong() }.toList())
+      is IntArray -> Struct(schema).put(LONG_LIST, value.map { s -> s.toLong() }.toList())
+      is LongArray -> Struct(schema).put(LONG_LIST, value.toList())
+      is FloatArray -> Struct(schema).put(FLOAT_LIST, value.map { s -> s.toDouble() }.toList())
+      is DoubleArray -> Struct(schema).put(FLOAT_LIST, value.toList())
+      is BooleanArray -> Struct(schema).put(BOOLEAN_LIST, value.toList())
+      is Array<*> -> asList(value.toList(), value::class.java.componentType.kotlin)
+      is Iterable<*> -> {
+        val elementTypes = value.map { it?.javaClass?.kotlin }.toSet()
```

Comment:
Is it possible to have an empty list here? If it is, I wonder if we getting empty set here? I didn't find a test covering this.

Reply:
Very good point, fix is coming.

Reply:
Fixed in https://github.com/neo4j/neo4j-kafka-connector/pull/166/commits/c6abde4f0aeb28238e974db669b71a4388aa7c3e.